engines/illusions/metaengine.cpp

```diff
@@ -75,6 +82,87 @@ void IllusionsMetaEngine::removeSaveState(const char *target, int slot) const {
 	g_system->getSavefileManager()->removeSavefile(fileName);
 }
 
+Common::KeymapArray IllusionsMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace Illusions;
+
+	Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "illusions-default", _("Default keymappings"));
+	Keymap *gameKeyMap = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts", _("Game keymappings"));
+
+	Common::Action *act;
+
+	act = new Action(kStandardActionLeftClick, _("Left click"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	act->addDefaultInputMapping("JOY_A");
+	act->addDefaultInputMapping("RETURN");
+	engineKeyMap->addAction(act);
+
+	act = new Action(kStandardActionRightClick, _("Right click"));
+	act->setRightClickEvent();
+	act->addDefaultInputMapping("MOUSE_RIGHT");
+	act->addDefaultInputMapping("JOY_B");
+	act->addDefaultInputMapping("BACKSPACE");
+	engineKeyMap->addAction(act);
+
+	act = new Action("CRSRUP", _("Move cursor up"));
+	act->setCustomEngineActionEvent(kActionCursorUp);
+	act->addDefaultInputMapping("JOY_UP");
+	act->addDefaultInputMapping("UP");
+	act->allowKbdRepeats();
+	gameKeyMap->addAction(act);
+
+	act = new Action("CRSRDOWN", _("Move cursor down"));
+	act->setCustomEngineActionEvent(kActionCursorDown);
+	act->addDefaultInputMapping("JOY_DOWN");
+	act->addDefaultInputMapping("DOWN");
+	act->allowKbdRepeats();
+	gameKeyMap->addAction(act);
+
+	act = new Action("CRSRLEFT", _("Move cursor left"));
+	act->setCustomEngineActionEvent(kActionCursorLeft);
+	act->addDefaultInputMapping("JOY_LEFT");
+	act->addDefaultInputMapping("LEFT");
+	act->allowKbdRepeats();
+	gameKeyMap->addAction(act);
+
+	act = new Action("CRSRRIGHT", _("Move cursor right"));
+	act->setCustomEngineActionEvent(kActionCursorRight);
+	act->addDefaultInputMapping("JOY_RIGHT");
+	act->addDefaultInputMapping("RIGHT");
+	act->allowKbdRepeats();
+	gameKeyMap->addAction(act);
+
+	act = new Action("ABORT", _("Abort"));
+	act->setCustomEngineActionEvent(kActionAbort);
+	act->addDefaultInputMapping("ESCAPE");
+	act->addDefaultInputMapping("JOY_X");
+	gameKeyMap->addAction(act);
+
+	act = new Action("SKIP", _("Skip"));
+	act->setCustomEngineActionEvent(kActionSkip);
+	act->addDefaultInputMapping("SPACE");
+	gameKeyMap->addAction(act);
+
+	act = new Action("INVENTORY", _("Open inventory"));
+	act->setCustomEngineActionEvent(kActionInventory);
+	act->addDefaultInputMapping("i");
+	act->addDefaultInputMapping("JOY_Y");
+	gameKeyMap->addAction(act);
+
+	act = new Action("ENCHEAT", _("Enable cheat mode"));
+	act->setCustomEngineActionEvent(kActionCheatMode);
+	act->addDefaultInputMapping("F1");
+	act->addDefaultInputMapping("JOY_LEFT_SHOULDER");
```

Comment:
Hmmm.. What does cheat mode do for this engine? I feel odd about accidentally cheating in a game by just trying out the buttons on the gamepad.

Reply:
The user has to input a certain keyword then only is he able to activate it.

Reply:
Alright - I checked that out an it makes sense. The one thing I would be interested in is a debugger console command to activate the cheat as well. That would probably fall out of scope for this PR.